fiatconnect-api.md

```diff
@@ -383,105 +399,201 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+  greater than four hours (14400 seconds) ahead of the `issued-at` field, the server MUST fail the request.
 
-The `sub`, or *subject* claim represents the user's Celo network address. This is a required claim. If it is missing, the server MUST respond
-to the client with an HTTP `400` error. This claim SHOULD be interpreted and used within server endpoints as a unique user identifier, and as the
-source of/destination for crypto funds during transfers. The address should be formatted as 42 hexadecimal characters prefixed by the string literal
-`0x`. If the `sub` claim does not match this format, the server MUST respond to the client with an HTTP `400` error.
+If all fields are acceptable, and the message is formatted correctly, the server MUST authenticate the user's session, and return an HTTP `200`.
+The server MUST attach the fields from the signed message to the session internally, in order to access data about the user and their session in other endpoints.
 
-##### 3.2.1.2.2. `"iss"` Claim
+Once a server authenticates a session using a particular nonce, the server MUST NOT allow any other sessions to be created using that nonce, until `expiration-time` of the session created
+using that nonce has been reached. The implementation of this is up to the server, but will require storing some global state about nonces in order to check if a client is attempting
+to sign-in using a nonce that is temporarily forbidden. Careful consideration should be taken to make sure that nonce management works correctly when scaling server resources.
 
-The `iss`, or *subject* claim represents the user's Celo address public key, dervied from their private key. This is a required claim.
-The public key MUST be in compressed form (see [here](https://docs.ethers.io/v5/api/utils/signing-key/) for more details), and exactly 33 bytes
-long.
+###### 3.2.1.5.3.2. Failure
 
-If it is missing, the server MUST respond to the client with an HTTP `400` error. The server MUST use this claim to verify the JWT signature.
-Since the address provided in the `"sub"` claim may not correspond to the public/private keypair used to sign and verify the JWT, the
-server MUST validate that the signature does indeed correspond to the address provided in the `"sub"` claim.
+On failure, the server MUST return an HTTP `401` error, along with an error code that details exactly why the request failed.
 
-##### 3.2.1.2.3. `"exp"` Claim
+###### 3.2.1.5.3.2.1 `InvalidSignature`
 
-The `exp`, or *expiration time* claim represents the time until which the provided JWT should be accepted by the server. The `exp` claim should be provided
-as a numeric timestamp, defined as the number of seconds since Epoch. The `exp` claim MUST be present for all endpoints *except* the following:
+If the signature is invalid or doesn't match the included `address` field, the server MUST return an `InvalidSignature` error.
```

Comment:
might be worth explaining that the included `address` field might not "match" the signature for contract-owned accounts